iothub/device/src/Transport/RetryDelegatingHandler.cs

```diff
@@ -640,6 +640,13 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
         /// </summary>
         private async Task EnsureOpenedAsync(CancellationToken cancellationToken)
         {
+            // If this object has already been disposed, we will throw an exception indicating that.
```

Comment:
nit: maybe add a note specifying that this is because the current behavior doesn't allow open -> close -> open